tests/models/test_taskinstance.py

```diff
@@ -481,18 +482,51 @@ def task_function(ti):
         ti.refresh_from_db()
         ti.state == state
 
-    def test_task_retry_wipes_next_fields(self, session, dag_maker):
+    @pytest.mark.parametrize(
+        "state",
+        [State.FAILED, State.SKIPPED, State.SUCCESS, State.UP_FOR_RESCHEDULE, State.UP_FOR_RETRY],
+    )
+    def test_task_wipes_next_fields(self, session, state, dag_maker):
         """
         Test that ensures that tasks wipe their next_method and next_kwargs
-        fields when they are queued for retry after a failure.
+        when they go into a state of FAILED, SKIPPED, SUCCESS, UP_FOR_RESCHEDULE, or UP_FOR_RETRY.
         """
 
-        with dag_maker('test_mark_failure_2'):
-            task = BashOperator(
-                task_id='test_retry_handling_op',
-                bash_command='exit 1',
-                retries=1,
-                retry_delay=datetime.timedelta(seconds=2),
+        def failure():
+            raise AirflowException
+
+        def skip():
+            raise AirflowSkipException
+
+        def success():
+            return None
+
+        def reschedule():
+            reschedule_date = timezone.utcnow()
+            raise AirflowRescheduleException(reschedule_date)
+
+        _retries = 0
+        _retry_delay = datetime.timedelta(seconds=0)
+
+        if state == State.FAILED:
+            _python_callable = failure
+        elif state == State.SKIPPED:
+            _python_callable = skip
+        elif state == State.SUCCESS:
+            _python_callable = success
+        elif state == State.UP_FOR_RESCHEDULE:
+            _python_callable = reschedule
+        elif state in [State.FAILED, State.UP_FOR_RETRY]:
+            _python_callable = failure
+            _retries = 1
+            _retry_delay = datetime.timedelta(seconds=2)
+
+        with dag_maker("test_deferred_method_clear"):
+            task = PythonOperator(
+                task_id="test_deferred_method_clear_task",
+                python_callable=_python_callable,
```

Comment:
```suggestion
        def run(state):
            if state == State.FAILED:
                raise AirflowException
            if state == State.SKIPPED:
                raise AirflowSkipException
            if state == State.UP_FOR_RESCHEDULE:
                raise AirflowRescheduleException(timezone.utcnow())
            return None  # SUCCESS

        _retries = 0
        _retry_delay = datetime.timedelta(seconds=0)

        if state in [State.FAILED, State.UP_FOR_RETRY]:
            _retries = 1
            _retry_delay = datetime.timedelta(seconds=2)
            
         with dag_maker("test_deferred_method_clear"):
            task = PythonOperator(
                task_id="test_deferred_method_clear_task",
                python_callable=run,
                op_args=[state],
```

Do the `retries` and `retry_delay` differences actually matter?

Reply:
@ReadytoRocc, I'm going to merge this for 2.2.1, but can you follow up with refactoring the test when you get the chance? Thanks.

Reply:
@jedcunningham thanks for the the suggestions. Will do.

Reply:
`Do the retries and retry_delay differences actually matter?`
@jedcunningham I believe it would only matter for the following cases:

1. `State.FAILED` needs `retries` set to `0` so that task goes into a state of failed.
2.  `State.UP_FOR_RETRY` needs `retries` set to `1` so that task goes into a state of retry.

I believe we can set `_retry_delay` once, default `_retries = 0`, and then only set `_retries = 1`, if `State.UP_FOR_RETRY`.

Reply:
@jedcunningham - PR for refactored tests: https://github.com/apache/airflow/pull/19194